specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,186 @@
+# Tree View + Tree View Item
+
+## Overview
+
+There are two components being presented as part of this specification - tree view, and tree view item. 
+
+As defined by the W3C:
+> A tree view widget presents a hierarchical list. Any item in the hierarchy may have child items, and items that have children may be expanded or collapsed to show or hide the children. For example, in a file system navigator that uses a tree view to display folders and files, an item representing a folder can be expanded to reveal the contents of the folder, which may be files, folders, or both.
+
+### Background
+
+Currently, a react implementation exists which supports a `defaultExpanded` value which sets the default expanded or collapsed state. In the event that the `defaultExpanded` value is false, but there is a selected child the tree will expand automatically. [Issue #2597](https://github.com/microsoft/fast-dna/issues/2597) requests an expanded prop to fully control the expanded/collapsed state.
+
+### Use Cases
+
+Primarily used to present a hierarchical list such as a file system. A tree view can also be served as a flat list.
+  
+### Features
+
+- **Selection:** A tree view can support either single select or multi-select behavior for nodes. **I would propose that for the initial spec we start with supporting single select only.*
+
+
+- **Drag & Drop support:** Often, as is in the case of a "file system" example, drag and drop support is desired. This does present [challenges](#drag-&-drop).
+
+### Risks and Challenges
+
+#### Nesting:
+
+Infinite nesting combined with the current visual design presents a styling challenge. Consider a scenario where we have the following tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        Parent Node && Root node
+        <fast-tree-item slot="child-nodes">Nested node one</fast-tree-item slot="child-nodes">
+        <fast-tree-item slot="child-nodes">Nested node two</fast-tree-item>
```

Comment:
`child-nodes` or `children` ?

Reply:
Maybe even better as `items` since it's not really child nodes, but tree view items.

Reply:
Idea: In the constructor of the `fast-tree-item` we can write the `slot` value so that consumers do not have to assign it. This will allow them to leave that out of their html and ensure that tree items always project to the correct slot.

Reply:
I'd be fine with either here in terms of naming. I do think it depends on what we want to land universally for *most* things here. I think both `fast-tree-view` and `fast-tree-item` should use the same naming here. Technically, I envision that a user could pass a <div> with the role of "treeitem" to either of these and it would still work. As long as it's in the slot and has the role of `treeitem` I'd expect everything to continue working.